`kedro-datasets`: Nightly build failure

GitHub Actions workflow [Run nightly tests on all plugins #299](https://github.com/kedro-org/kedro-plugins/actions/runs/9637428241) failed.

> E   RuntimeError: Failed to import transformers.pipelines because of the following error (look up to see its traceback):
E   cannot import name 'split_torch_state_dict_into_shards' from 'huggingface_hub' (/workspace/kedro-plugins/.venv/lib/python3.10/site-packages/huggingface_hub/__init__.py)

I can reproduce this when I run `pip install .[test]`. This does not happen when I run `pip install .[huggaceface]`. It could be some secondary dependencies issue.

--- a/kedro-datasets/pyproject.toml
+++ b/kedro-datasets/pyproject.toml
@@ -179,6 +179,7 @@ docs = [
 
 # Test requirements
 test = [
+    "accelerate<0.32", # Temporary pin
     "adlfs~=2023.1",
     "bandit>=1.6.2, <2.0",
     "behave==1.2.6",

